src/main/java/hudson/remoting/Launcher.java

```diff
@@ -127,17 +121,25 @@ public void setTextMode(boolean b) {
         System.out.println("Running in "+mode.name().toLowerCase(Locale.ENGLISH)+" mode");
     }
 
+    /**
+     * @deprecated use {@link #secret}, {@link #name}, {@link #urls}, {@link #webSocket}, {@link #tunnel},
+     * {@link #workDir}, {@link #internalDir}, and/or {@link #failIfWorkDirIsMissing} directly.
+     */
     @Option(name="-jnlpUrl",usage="instead of talking to the controller via stdin/stdout, " +
             "emulate a JNLP client by making a TCP connection to the controller. " +
-            "Connection parameters are obtained by parsing the JNLP file.")
+            "Connection parameters are obtained by parsing the JNLP file.", forbids = {"-direct", "-name", "-tunnel", "-url", "-webSocket"})
+    @Deprecated
     public URL agentJnlpURL = null;
 
-    @Option(name="-jnlpCredentials",metaVar="USER:PASSWORD",usage="HTTP BASIC AUTH header to pass in for making HTTP requests.")
+    @Option(name="-credentials",metaVar="USER:PASSWORD",aliases="-jnlpCredentials",usage="HTTP BASIC AUTH header to pass in for making HTTP requests.")
```

Comment:
`-credentials` in the other entrypoint. Since JNLP is an outdated concept, I think it's better to prefer `-credentials` as the main name for this argument.